### ISSUE TITLE
Crash with wildcards and method references

### ISSUE THREAD

**Issue by smillst** [typetools/checker-framework#8238]
```java
/**
 * <p>A generic method reference checked against a functional interface whose parameter type is a
 * wildcard-parameterized type. {@code Box<Tag<?>, Object>} fixes the {@code Fn} parameter to
 * {@code Tag<? extends Object>}, and the method reference {@code this::create} is generic in
 * {@code T}.
 *
 * <p>Inference reduces the constraint {@code Tag<? extends Object> <: Tag<T>} for the method
 * reference's formal parameter. The bound is false, so inference throws FalseBoundException and the
 * Checker Framework crashes in InvocationTypeInference.createB2MethodRef, reached from
 * BaseTypeVisitor.checkMethodReferenceAsOverride.
 *
 * <p>Run with -AconvertTypeArgInferenceCrashToWarning=false, which is what the Beam build sets;
 * otherwise the crash is reported as a type.argument.inference.crashed error instead.
 *
 * <p>Reduced from org.apache.beam.fn.harness.FnApiDoFnRunner, which is
 * {@code taggedOutputReceivers.computeIfAbsent(tag, this::createTaggedOutputReceiver)} with a
 * {@code Map<TupleTag<?>, OutputReceiver<?>>}. The crash needs no JDK types: {@code Box} stands in
 * for {@code Map} and {@code compute} for {@code computeIfAbsent}. The cast and the generic return
 * type at the original call site are not needed. The wildcard must come from the receiver's type
 * arguments; if {@code K} is instead inferred from the call's own argument, there is no crash.
 */
public class FnApiDoFnRunnerCrash {

  interface Fn<A, B> {
    B apply(A a);
  }

  static class Tag<T> {}

  static class Box<K, V> {
    V compute(K key, Fn<? super K, ? extends V> fn) {
      throw new UnsupportedOperationException();
    }
  }

  Box<Tag<?>, Object> box = new Box<>();

  <T> Object create(Tag<T> tag) {
    throw new UnsupportedOperationException();
  }

  <T> void get(Tag<T> tag) {
    box.compute(tag, this::create);
  }
}
```
